src/screens/discover/DiscoverDetailScreen.tsx

```diff
@@ -71,34 +61,76 @@ const requestHandler = async (props: RequestHandlerProps) => {
 };
 
 const DiscoverDetailScreen: React.FC = () => {
+    const query: string = useLoaderData() as string;
     const { profile, setProfile } = useProfileContext();
     const path = window.location.pathname
         .match(/\/\w+$/)
         ?.join()
         .slice(1);
-    const [data, setData] = useState<ShowData[] | null>(null);
     const [loading, setLoading] = useState<boolean>(true);
+    const storageItem = localStorage.getItem('streamabilityView');
+    const initialView = storageItem === 'grid' ? 'grid' : 'list';
+    const [viewState, setViewState] = useState<'list' | 'grid'>(initialView);
+    const [hash, setHash] = useState<number>(1);
+    const {
+        data,
+        setData,
+        loading: dataLoading,
+        moreToFetch,
+        refetch,
+    } = usePaginatedData({ query: query });
 
     useEffect(() => {
         if (path) requestHandler({ path: path, setState: setData, setLoading: setLoading });
     }, []);
 
-    const title = path ? PATHS[PATHS.findIndex((p) => p.path === path)].title : '';
+    const observer = useRef<IntersectionObserver | null>(null);
+    const loadMoreRef = useCallback(
+        (node: HTMLDivElement) => {
+            if (dataLoading) return;
+            if (observer.current) observer.current.disconnect();
+            observer.current = new IntersectionObserver((entries) => {
+                if (entries[0].isIntersecting && moreToFetch) {
+                    refetch();
+                }
+            });
+            if (node) observer.current.observe(node);
+        },
+        [dataLoading, moreToFetch, refetch]
+    );
+
+    const cards = useMemo(() => {
+        return (
+            <SearchResultCards
+                details={data}
+                viewState={viewState}
+                profile={profile}
+                setProfile={setProfile}
+            />
+        );
+    }, [data, hash, viewState, profile]);
 
-    // TODO: Create loader #839
+    // TODO: Create loader #839r
     if (loading) return <p>Loading...</p>;
 
     return (
         <div
             className='flex flex-col items-center w-full m-6'
             data-testid={`discover-details-${path}-screen`}
         >
-            <Typ variant='h4'>{title}</Typ>
-            <div className={'grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4'}>
-                {data?.map((item, i) => (
-                    <ShowCard key={i} details={item} profile={profile} setProfile={setProfile} />
-                ))}
-            </div>
+            <DetailScreen
+                query={`trending`}
```

Comment:
This should not be hardcoded. It should be using the `PATHS` object from line 18.

Reply:
fix path value so it shows the title bar and name.